File: Src/Server/Kloon.EmployeePerformance.Models/QuarterEvaluation/DataSourcePersonalEvaluateModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kloon.EmployeePerformance.Models.QuarterEvaluation
{
    public class DataSourcePersonalEvaluateModel
    {
        public bool IsAvaibleEvaluate { get; set; }

        public List<ProjectDataSourceModel> ProjectSource { get; set; }
        public List<QuarterDataSourceModel> QuarterSource { get; set; }
        public List<YearDataSourceModel> YearSource { get; set; }
        public List<QuarterEvaluationModel> DataSource { get; set; }
    }

    public class YearDataSourceModel
    {
        public int Id { get; set; }
        public string Value { get; set; }
    }

    public class QuarterDataSourceModel
    {
        public int Id { get; set; }
        public int YearId { get; set; }
        public string Value { get; set; }
    }

    public class ProjectDataSourceModel
    {
        public int Id { get; set; }
        public int YearId { get; set; }
        public int QuarterId { get; set; }
        public string Value { get; set; }
        public int ProjectId { get; set; }

    }
}

File: Src/Client/src/app/shared/models/datasource-personal-evaluate.model.ts
import { QuarterEvaluationModel } from "./quarter-evaluation.model";

export class DataSourcePersonalEvaluateModel {
    isAvaibleEvaluate: boolean;
    projectSource: ProjectDataSourceModel[];
    quarterSource: QuarterDataSourceModel[];
    yearSource: YearDataSourceModel[];
    dataSource: QuarterEvaluationModel[];
}

export class YearDataSourceModel {
    id: number;
    value: string;
}

export class QuarterDataSourceModel {
    id: number;
    yearId: number;
    value: string;
}

export class ProjectDataSourceModel {
    id: number;
    yearId: number;
    quarterId: number;
    value: string;
    projectId: number;
}